packages/webapp/pages/callback.tsx

```diff
@@ -25,8 +25,12 @@ function CallbackPage(): ReactElement {
         window.location.replace(`/reset-password?${search}`);
         return;
       }
-      postWindowMessage(eventKey, params);
-      broadcastMessage({ ...params, eventKey });
+
+      if (window.opener?.postMessage) {
+        postWindowMessage(eventKey, params);
+      } else {
+        broadcastMessage({ ...params, eventKey });
+      }
```

Comment:
@rebelchris we basically have two listeners now, although we will only trigger the one from broadcast channel only when the `postMessage` is `undefined`.

This is to make the facebook fix smallest possible and keep the existing system intact 🙏

Reply:
Why can't we just use broadcast everywhere then?

Reply:
Broadcast Channel doesn't allow cross-origin messaging. That would mean our preview deployments will stop working since preview deployments redirect is towards the main `app.daily.dev`.

Reply:
Hmm just feels very edge-casey and potentially introduce broken stuff we don't know what causes.
I'm happy to go for this, but we'll need to make sure everyone in web is notified and aware in case some weird issues arrise. also make sure Ido is onboard here.

Reply:
Yeah, this was the result of discussion with Ido in our 1-1 after trying to debug it further.

Reply:
Since we are running out of time and we just want to fix the Facebook for now.

Reply:
If only BroadcastChannel allows cross-origin, it would've been great to migrate. Unfortunately, I think the trade-off is too much. We heavily test on our preview deployments and we cannot, not have that. Hopefully we get to understand soon with the `window.postMessage` is `undefined` even when the `document.readyState` is already `complete`.

Reply:
cc @idoshamun

Reply:
Was the issue confirmed fixed with BC?

Reply:
Yes, locally. We can't test in in preview, since like mentioned the cross-origin won't allow it.

Reply:
I'm fine with this solution, Lee and I decided together to use it as a fallback. I actually think it's even better than just supporting FB flow in this case